packages/contracts/contracts/protocol/AssetProxy/MultiAssetProxy.sol

```diff
@@ -0,0 +1,249 @@
+                        mstore(64, 0x0000001055494e543235365f4f564552464c4f57000000000000000000000000)
+                        mstore(96, 0)
+                        revert(0, 100)
+                    }
+
+                    // Write `totalAmount` to memory
+                    mstore(100, totalAmount)
+
+                    // Load offset to `nestedAssetData[i]`
+                    let nestedAssetDataElementOffset := calldataload(add(nestedAssetDataContentsStart, i))
+
+                    // Load length of `nestedAssetData[i]`
+                    let nestedAssetDataElementContentsStart := add(nestedAssetDataElementOffset, 264)
+                    let nestedAssetDataElementLen := calldataload(sub(nestedAssetDataElementContentsStart, 32))
+
+                    // Revert if the `nestedAssetData` does not contain a 4 byte `assetProxyId`
+                    if lt(nestedAssetDataElementLen, 4) {
+                        // Revert with `Error("LENGTH_GREATER_THAN_3_REQUIRED")`
+                        mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
+                        mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
+                        mstore(64, 0x0000001e4c454e4754485f475245415445525f5448414e5f335f524551554952)
+                        mstore(96, 0x4544000000000000000000000000000000000000000000000000000000000000)
+                        revert(0, 100)
+                    }
+
+                    // Load AssetProxy id
+                    let assetProxyId := and(calldataload(nestedAssetDataElementContentsStart), 0xffffffff00000000000000000000000000000000000000000000000000000000)
+
+                    // To lookup a value in a mapping, we load from the storage location keccak256(k, p),
+                    // where k is the key left padded to 32 bytes and p is the storage slot
+                    mstore(132, assetProxyId)
+                    mstore(164, assetProxies_slot)
+                    let assetProxy := sload(keccak256(132, 64))
+                    
+                    // Revert if AssetProxy with given id does not exist
+                    if iszero(assetProxy) {
+                        // Revert with `Error("ASSET_PROXY_DOES_NOT_EXIST")`
+                        mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
+                        mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
+                        mstore(64, 0x0000001a41535345545f50524f58595f444f45535f4e4f545f45584953540000)
+                        mstore(96, 0)
+                        revert(0, 100)
+                    }
+    
+                    // Copy `nestedAssetData[i]` from calldata to memory
+                    calldatacopy(
+                        132,                                // memory slot after `amounts[i]`
+                        nestedAssetDataElementOffset,       // location of `nestedAssetData[i]` in calldata
```

Comment:
This does not take offset from outer assetData into account, nor the selectors. Should be `add(assetDataOffset, add(nestedAssetDataElementOffset, 4 + 4))` or `sub(nestedAssetDataElementContentsStart, 32)`.